llmv_routes.py

```diff
@@ -41,8 +42,44 @@ def generate_for_challenge():
         """Add a route to CTFd for generating text from a prompt."""
         log.info(f'Received text generation request from user "{get_current_user().name}" '
                  f'for challenge ID "{request.json["challenge_id"]}"')
+        
         challenge = LlmChallenge.query.filter_by(id=request.json['challenge_id']).first_or_404()
 
+        if 'generation_id' in request.json:
+            log.info("Found old generation id, using that")
+            llmv_generation = LLMVGeneration.query.filter_by(id=request.json['generation_id']).first_or_404()
+            if llmv_generation.status != "pending":
+                response = {'success': False, 'data': {'text': "This challenge is complete.", "id": -1}}
+                return jsonify(response)
+            if llmv_generation.account_id != get_current_user().id:
+                response = {'success': False, 'data': {'text': "This challenge is complete.", "id": -1}}
+                return jsonify(response)
+            if llmv_generation.challenge_id != challenge.id:
+                response = {'success': False, 'data': {'text': "This challenge is complete.", "id": -1}}
+                return jsonify(response)
+            history = llmv_generation.history
+            history = [h.json() for h in llmv_generation.history]
+            log.debug(f'Found history "{history}" ')
```

Comment:
Very optional suggestion, lazy evaluation is suggested for logging statements

```suggestion
            log.debug('Found history %s ', history)
```